cmd/release.go

```diff
@@ -88,6 +130,137 @@ func newCmdRelease() *cobra.Command {
 	return cmd
 }
 
+func confirmReleasing(listedRelease bool) (bool, error) {
+	var confirmMsg string
+
+	if listedRelease {
+		shared.Logger.Println("Creating a listed release makes your app available on Deta Discovery for anyone to install and use.")
+		confirmMsg = "Are you sure you want to release this app publicly on Discovery? (y/n)"
+	} else {
+		shared.Logger.Println("Releasing makes your app available via a unique link for others to install and use.")
+		confirmMsg = "Are you sure you want to release this app to others? (y/n)"
+	}
+	shared.Logger.Printf("If you only want to use this app yourself, use your Builder instance instead.\n\n")
+
+	continueReleasing, err := confirm.Run(confirmMsg)
+	if err != nil {
+		return false, fmt.Errorf("problem while trying to get confirmation to continue releasing this project, %w", err)
+	}
+
+	return continueReleasing, nil
+}
+
+func validatePromptValue(value string, min int, max int) error {
+	if len(value) < min {
+		return fmt.Errorf("must be at least %v characters long", min)
+	}
+
+	if len(value) > max {
+		return fmt.Errorf("must be at most %v characters long", max)
+	}
+
+	return nil
+}
+
+func validateAppName(value string) error {
+	return validatePromptValue(value, 4, 12)
+}
+
+func validateAppDescription(value string) error {
+	return validatePromptValue(value, 4, 69)
+}
+
+func compareDiscoveryData(discoveryData *sharedTypes.DiscoveryFrontmatter, latestRelease *api.Release, projectDir string) error {
+	if latestRelease.Discovery.ContentRaw != "" && !reflect.DeepEqual(latestRelease.Discovery, discoveryData) {
+		modTime, err := discovery.GetDiscoveryFileLastChanged(projectDir)
+		if err != nil {
+			shared.Logger.Println(styles.Errorf("%s Failed to check if local Discovery data is outdated: %v", emoji.ErrorExclamation, err))
+			return err
+		}
+
+		parsedTime, err := time.Parse(time.RFC3339, latestRelease.ReleasedAt)
+		if err != nil {
+			shared.Logger.Println(styles.Errorf("%s Failed to check if local Discovery data is outdated: %v", emoji.ErrorExclamation, err))
+			return err
+		}
+
+		if modTime.Before(parsedTime) {
+			shared.Logger.Print("\nWarning: your local Discovery data is different from the latest release's Discovery data.\n\n")
```

Comment:
Context: we compare the created date of the latest release with the modified date of the Discovery file to figure out if we should warn the user about out of sync data:

- if the Discovery file was last modified _before_ the creation of the latest release we assume it is outdated as the latest release now contains different data and we warn them
- if the Discovery file was last modified after the creation of the latest release we assume the dev changed the file knowingly after having previously released through the UI and we don't need to warn them

Reply:
I don't get why this is necessary. Is there a way for users to modify the discovery file outside of the space release command ?

Reply:
yeah, part of this project was that you will be able to do it from Builder when creating a release:

![CleanShot 2023-04-04 at 15 53 35@2x](https://user-images.githubusercontent.com/51766171/229814966-2e4516c7-fdf1-44fe-9b85-245a9f2321d8.png)